test/common/user_permissions.go

```diff
@@ -0,0 +1,227 @@
+package common
+
+import (
+	goctx "context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/go-querystring/query"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	projectv1 "github.com/openshift/api/project/v1"
+	v1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	"strings"
+
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"testing"
+)
+
+const (
+	openshiftAuthenticationNamespace = "openshift-authentication"
+	openshiftOAuthRouteName          = "oauth-openshift"
+
+	pathProjects = "api/kubernetes/apis/project.openshift.io/v1/projects"
+	pathFusePods = "api/kubernetes/api/v1/namespaces/redhat-rhmi-fuse/pods"
+
+	expectedRhmiDeveloperNamespace    = "redhat-rhmi-fuse"
+	expectedRhmiDeveloperProjectCount = 1
+	expectedFusePodCount              = 6
+)
+
+// struct used to create query string for fuse logs endpoint
+type LogOptions struct {
+	Container string `url:"container"`
+	Follow    string `url:"follow"`
+	TailLines string `url:"tailLines"`
+}
+
+func TestIntegreatlyUserPermissions(t *testing.T, ctx *TestingContext) {
+	// get the RHMI custom resource to check what storage type is being used
+	rhmi := &v1alpha1.RHMI{}
+	ns := fmt.Sprintf("%soperator", namespacePrefix)
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: installationName, Namespace: ns}, rhmi); err != nil {
+		t.Fatal("error getting RHMI CR:", err)
+	}
+	// get console master url
+	masterURL := rhmi.Spec.MasterURL
+
+	// get oauth route
+	oauthRoute := &v1.Route{}
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: openshiftOAuthRouteName, Namespace: openshiftAuthenticationNamespace}, oauthRoute); err != nil {
+		t.Fatal("error getting Openshift Oauth Route: ", err)
+	}
+
+	// get rhmi developer user tokens
+	rhmiDevUserToken, err := resources.GetUserOpenshiftToken(oauthRoute.Spec.Host, masterURL, resources.DefaultIDP, "test-user01", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	// get projects for rhmi developer
+	rhmiDevfoundProjects, err := ocGetProjects(masterURL, rhmiDevUserToken)
+	if err != nil {
+		t.Fatalf("error occured while getting user projects : %v", err)
+	}
+
+	// check if projects are as expected for rhmi developer
+	for projectCount, p := range rhmiDevfoundProjects.Items {
+		t.Log(fmt.Sprintf("found rhmi-developer project - %s", p.Name))
+		if projectCount >= expectedRhmiDeveloperProjectCount {
+			t.Fatal(fmt.Sprintf("test failed - project count for rhmi-developer exceeded expected 1"))
+		}
+		if p.Name != expectedRhmiDeveloperNamespace {
+			t.Fatal(fmt.Sprintf("test failed - found project for rhmi-developer does not match expected : %s", expectedRhmiDeveloperNamespace))
+		}
+	}
+	t.Log("test-passed - found projects for rhmi-developer are as expected")
+
+	// get dedicated admin token
```

Comment:
Wouldn't it be better to split the developer and dedicated admin test permissions in two different tests? They are one manual tests just to simplify the manual execution, but I would prefer to have two automated tests because they are not dependent from each other